pkg/query/api/v1.go

```diff
@@ -265,14 +265,27 @@ func (api *API) queryRange(r *http.Request) (interface{}, []error, *apiError) {
 
 	step, err := parseDuration(r.FormValue("step"))
 	if err != nil {
-		return nil, nil, &apiError{errorBadData, err}
+		return nil, nil, &apiError{errorBadData, errors.Wrap(err, "param step")}
 	}
 
 	if step <= 0 {
 		err := errors.New("zero or negative query resolution step widths are not accepted. Try a positive integer")
 		return nil, nil, &apiError{errorBadData, err}
 	}
 
+	maxSourceResolution := step / 5 // By default fit at least 5 samples between steps.
+	if val := r.FormValue("max_source_res"); val != "" {
```

Comment:
Nit: can we name this `max_source_resolution`? I have a bad experience with naming the same thing with 2 slightly different names, it increases confusion level a bit.

Reply:
makes sense!